src/main/java/io/tiledb/java/api/Attribute.java

```diff
@@ -292,19 +292,22 @@ public void setFillValue(NativeArray value, BigInteger size) throws TileDBError
    * <p>Applicable to var-sized attributes.
    *
    * @param value The fill value
+   * @param size The fill value size
    * @throws TileDBError
    */
   public void setFillValue(Object value) throws TileDBError {
-    NativeArray array = new NativeArray(ctx, this.type.getNativeSize(), this.type.javaClass());
+    int size;
+
+    if (value instanceof String) size = ((String) value).length();
+    else size = this.type.getNativeSize();
```

Comment:
I think we also need to handle var length and fix-length greater than one. If its not a string we need to check for arrays like `int[]` and `long[]`, etc. If the user passes an array or a string, and the attribute has a fixed size > 1 (`getCellValNum` but not `isVar()`) we should make sure the length/element count lines up. 

Both of the above I think are what we need to do to have robust support here, but please validate my logic and assumptions.